proto/proto/block_header.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+package block_header;
+
+import "digest.proto";
+
+message BlockHeader {
+    /// the hash of the previous blocks header.
+    digest.Digest prev_hash = 1;
+    /// a unique sequential number of the current block.
+    uint64 block_num = 2;
+    /// a commitment to an MMR of the entire chain where each block is a leaf.
+    digest.Digest chain_root = 3;
+    /// a commitment to account database.
+    digest.Digest account_root = 4;
+    /// a commitment to the nullifier database.
+    digest.Digest nullifier_root = 5;
+    /// a commitment to all notes created in the current block.
+    digest.Digest note_root = 6;
+    /// a commitment to a set of transaction batches executed as a part of this block.
+    digest.Digest batch_root = 7;
+    /// a hash of a STARK proof attesting to the correct state transition.
+    digest.Digest proof_hash = 8;
+    /// specifies the version of the protocol.
+    uint64 version = 9;
+    /// the time when the block was created.
+    uint64 timestamp = 10;
```

Comment:
`u32` values are probably enough for these fields.

Reply:
changed, although I disagree this is a good idea, this is basically setting up the response to have a Y2K problem

Reply:
I guess the question is what we store here. My initial assumption was that it would be seconds and then `u32` should be plenty (if I did my math correctly, that should give us 130+ years). But if we want to store milliseconds here, then we should go to `u64`.

Reply:
The epoch is also important. The UNIX epoch is `January 1, 1970`, that is 53 years, so that leaves 83years if seconds are used. It should be plenty, but I don't see what we gain with the change.

Reply:
Ah - for w/e reason I thought we'd start with 0, but we probably should just use standard epochs. In that case, yes, let's go to `u64`. I would still decrease version to `u32` - unless there is a good reason to keep it at `u64`?

Reply:
changed timestamp back to u64, changed version to u32